lib/types/SSI.types.ts

```diff
@@ -54,16 +62,200 @@ export interface Issuer {
   [x: string]: unknown;
 }
 
-export interface Credential {
-  '@context': string[];
-  id: string;
-  type: string[];
+export interface InternalCredential {
+  getAudience(): string | undefined;
+
+  getBaseCredential(): InternalCredentialBase;
+
+  getContext(): string[] | string;
+
+  getExpirationDate(): string | undefined;
+
+  getId(): string | undefined;
+
+  getIssuer(): unknown;
+
+  getIssuanceDate(): string | undefined;
+
+  getJti(): string | undefined;
+
+  getType(): string;
+
+  [x: string]: unknown;
+}
+
+export class InternalCredentialBase {
+  '@context': string[] | string;
+  credentialStatus?: CredentialStatus;
   credentialSubject: CredentialSubject;
-  issuer: string | Issuer;
-  issuanceDate: string;
+  description?: string;
   expirationDate?: string;
+  id: string;
+  issuanceDate: string;
+  issuer: unknown;
+  name?: string;
+  type: string[];
+
+  [x: string]: unknown;
+}
+
+export class InternalCredentialJWT implements InternalCredential {
+  /**
+   * aud MUST represent (i.e., identify) the intended audience of the verifiable presentation
+   * (i.e., the verifier intended by the presenting holder to receive and verify the verifiable presentation).
+   */
+  aud?: string;
+
+  /**
+   * MUST represent the expirationDate property, encoded as a UNIX timestamp (NumericDate).
+   */
+  exp?: string;
+
+  /**
+   * MUST represent the issuer property of a verifiable credential or the holder property of a verifiable presentation.
+   */
+  iss: string;
+
+  /**
+   * MUST represent the id property of the verifiable credential or verifiable presentation.
+   */
+  jti?: string;
+
+  /**
+   * MUST represent issuanceDate, encoded as a UNIX timestamp (NumericDate).
+   */
+  nbf?: string;
+
+  /**
+   * MUST represent the id property contained in the verifiable credential subject.
+   */
+  sub?: string;
+
+  vc: InternalCredentialBase;
+
+  [x: string]: unknown;
+
+  getBaseCredential(): InternalCredentialBase {
+    return this.vc;
+  }
+
+  getContext(): string[] | string {
+    return this.vc['@context'];
+  }
+
+  getType(): string {
+    return 'jwt';
+  }
+
+  getAudience(): string | undefined {
+    return this.aud;
+  }
+
+  getExpirationDate(): string | undefined {
+    return this.exp;
+  }
+
+  getId(): string | undefined {
+    return this.sub;
```

Comment:
I think we can make some enhancements by also checking the vc for values that could be present there next to the the claims in the JWT itself:

See : 
```
If exp is present, the UNIX timestamp MUST be converted to an [ RFC3339 XMLSCHEMA11-2 ] date-time, and MUST be used to set the value of the expirationDateproperty of credentialSubject of the new JSON object.
If iss is present, the value MUST be used to set the issuer property of the new verifiable credential JSON object or the holder property of the new verifiable presentation JSON object.
If nbf is present, the UNIX timestamp MUST be converted to an [ RFC3339 XMLSCHEMA11-2 ] date-time, and MUST be used to set the value of the issuanceDatepropertyof the new JSON object.
If sub is present, the value MUST be used to set the value of the id property of credentialSubject of the new JSON object.
If jti is present, the value MUST be used to set the value of the id property of the new JSON object.
````

So basically if these values are present in the JWT they should be used. That suggests they could also already be present in the VC object, more conforming to the normal VC model. Right now we are assuming these claims to always be present in the JWT and simply only use them. See the (de)serialization chapters of the VC-datamodel